src/ui/qgsmssqlnewconnectionbase.ui

```diff
@@ -231,50 +258,69 @@ Untick save if you don't wish to be the case.</string>
         </property>
        </widget>
       </item>
-      <item row="2" column="0" colspan="2">
-       <widget class="QCheckBox" name="cb_geometryColumns">
+      <item row="5" column="0" colspan="2">
+       <widget class="QCheckBox" name="mCheckNoInvalidGeometryHandling">
         <property name="toolTip">
-         <string>If checked, only tables which are present in the &quot;geometry_columns&quot; metadata table will be available. This speeds up table scanning, but requires users to manually manage the geometry_columns table and ensure that layers are correctly represented in the table.</string>
+         <string>If checked, all handling of records with invalid geometry will be disabled. This speeds up the provider, however, if any invalid geometries are present in a table then the result is unpredictable and may include missing records. Only check this option if you are certain that all geometries present in the database are valid, and any newly added geometries or tables will also be valid.</string>
         </property>
         <property name="text">
-         <string>Only look in the geometry_columns metadata table</string>
+         <string>Skip invalid geometry handling</string>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
-      <item row="6" column="0" colspan="2">
-       <widget class="QPushButton" name="btnConnect">
-        <property name="text">
-         <string>Test Connection</string>
+      <item row="7" column="0" colspan="2">
+       <widget class="Line" name="line">
```

Comment:
Can you remove this line? It's not a UI element we commonly use in QGIS, and the group box along should be sufficient to provide the visual seperation